### ISSUE TITLE
Clippy wrongly suggests function as dead code when used in a closure

### ISSUE THREAD

**Issue by gsayson** [rust-lang/rust-clippy#9582]
### Summary

When calling a function _A_ from a closure defined in a function _B_, Clippy registers function _A_ as dead code, when it is actually being used.

### Lint Name

dead_code

### Reproducer

I tried this code:

```rust
pub async fn create_account(username: impl ToString, password: impl ToString) {
	let (username, password) = (username.to_string(), password.to_string());
	tokio::task::spawn_blocking(|| create_account_blocking(username, password)).await.unwrap()
}

#[inline(always)]
fn create_account_blocking(username: String, password: String) {
	let pwd_hash = Hasher::new()
		.with_salt(Salt::random(32))
		.with_password(password.clone())
		.hash()
		.unwrap();
	let db = sled::open("./zauth").unwrap();
	db.insert(username.as_bytes(), password.as_bytes()).unwrap();
}
```

I saw this happen:

```
warning: function `create_account_blocking` is never used
  --> zentral-server\src\auth.rs:33:4
   |
33 | fn create_account_blocking(username: String, password: String) {
   |
```

I expected it to compile without any warnings.

### Version

```text
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: x86_64-pc-windows-msvc
release: 1.64.0
LLVM version: 14.0.6
```

**Comment by Alexendoo** [comment]
Is `create_account` used? Unused functions aren't counted as a use for the purpose of the dead_code warning

**Comment by gsayson** [comment]
Ah, so that was the issue!
No, `create_account` wasn't used.
Thanks for your help!
